### PR TITLE
Created DropDownWithLabel component and Fixed dropdown styling to match Figma

### PR DESCRIPTION
**Pull request by kea-roy**
## Summary 

Created DropDownWithLabel component and revamped all pages with dropdown component to avoid repetitive code. Fixed dropdown styling to better match Figma. This pull request is a continuation of pull requests #352 and #353, specifically to resolve https://github.com/cornell-dti/cu-apts/pull/353#pullrequestreview-1956440909 

## Changes
- [x] Created DropDownWithLabel component, with label styling and mobile support 
- [x] Replaced DropDown components with DropDownWithLabel components on all pages.
- [x] Adjusted dropdown margins and font sizes to match Figma
- [x] Changed dropdown arrow icon to match Figma
- [x] Added dropdown arrow icon animation when the menu is opened 

### Test Plan 

Checked every webpage that had a dropdown to make sure the styling was correct on both desktop and mobile screens.
Screenshots of the webpage on desktop and mobile screens are shown below

Screenshot 2024-04-09 at 18 00 32
Screenshot 2024-04-09 at 18 03 38
Screenshot 2024-04-09 at 18 04 04
Screenshot 2024-04-09 at 18 04 40
Screenshot 2024-04-09 at 18 00 43

### CODE REVIEW

**Comment by kea-roy** [comment]
## Commit Changes

1. Improved documentation for DropDownWithLabel

File: frontend/src/components/utils/DropDown.tsx [hunk 1]
```diff
@@ -27,8 +27,22 @@
...
 import { Button, Menu, MenuItem } from '@material-ui/core';
 import { makeStyles } from '@material-ui/styles';
-import { ArrowDropDown, ArrowDropUp } from '@material-ui/icons';
-import SvgIcon from '@material-ui/core/SvgIcon';
+
+const expandArrow = (direction: boolean) => {
+  return (
+    <div style={{ paddingLeft: '10px', transform: direction ? 'rotate(180deg)' : 'none' }}>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
```
Comment (CasperL1218) [inline frontend/src/components/utils/DropDown.tsx:35]: I am not sure if this is necessary, but I think creating a local file for the icon/svg in the asset folder and referencing the icon from the asset would be:
1) More flexible for future modification
2) Easier for inspection 
3) Conforming to the general code pattern

**Review by CasperL1218 — CHANGES_REQUESTED** [review]
Your implementation and refinement of the overall dropdown mechanism are fully robust and complete after multiple iterations. The refactoring of the dropDownLabel component makes the code even more organized, which is really helpful. 

There are two small things that are not directly related to the dropdown mechanism, but you might want to look into:
(1). The referencing to the SVG file arrow icon is using a "http" path. I believe it could be replaced with creating a SVG file for the icon in the asset folder for reference to both offer an easier inspection and align with the code base practices.
(2). The import of {Review} in backend/scripts/add_price_and_bedroom_attributes.ts might not be necessary and can be removed unless you are using it in later implementation.

File: frontend/src/components/utils/DropDown.tsx (hunk 1 above)
Comment (kea-roy) [reply frontend/src/components/utils/DropDown.tsx]: Thanks for pointing this out! I've updated my code to fix this code format inconsistency in commit 89e93ba.

**Comment by kea-roy** [comment]
## Commit Changes
1. Added animation for dropdown
2. Adjusted code formatting for DropDown.tsx
3. Moved dropdown arrow svg into a separate file and linked it via an img element.

**Pull request merged**